Datasets, state files and scripts for DMS paper

julesvidal opened the pull request:
Dear Julien, 
This PR contains the state files, data, scripts and documentation for the experiments from the DMS paper.
Additionally, I modified the `run.py` script in `tests/`, to be able to specify a resolution factor from the commandline, in order to produce high-res pictures from pvsm files.
I'll prepare some screenshots of the paraview states for the website. 
Cheers,
Jules

julesvidal committed:
modified run.py to change resolution of screenshots

julesvidal committed:
progress

julesvidal committed:
Datasets, state files and scripts for DMS paper

julesvidal committed:
added darkSky data file

julesvidal committed:
fixed default value for resolution factor in run.py

julesvidal committed:
entry for darksky dataset in the readme

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
 }
+
+{
+  "name": "at.vti",
```
could you please insert these items at the right spot in the file?
(the entries are supposed to be in alphabetical order of their name).

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+{
+  "name": "at.vti",
+  "description": "Quantum chemistry dataset. Simulation of the electronic density in the Adenine-Thymine molecular complex",
+  "acknowledgement": "",
```
"acknowledgement": "Roberto Alvarez Boto",

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+  "name": "ds14_scivis_0128_e4_dt04_1.0000.vtp",
+  "description": "Point cloud dataset, simulation of the density of dark matter in the universe",
+  "acknowledgement": "",
+  "url": ""
```
please check mathieu's paper to get the URL to the right scivis contest: https://arxiv.org/pdf/2107.07789

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+{
+  "name": "ds14_scivis_0128_e4_dt04_1.0000.vtp",
+  "description": "Point cloud dataset, simulation of the density of dark matter in the universe",
+  "acknowledgement": "",
```
"IEEE SciVis Contest 20xx"
xx to be determined by looking at mathieu's paper: https://arxiv.org/pdf/2107.07789

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+{
+  "name": "ferility.vtu",
+  "description": "2d triangulation of the *fertility* dataset. Scanning and reconstruction.",
+  "acknowledgement": "",
```
"acknowledgement": "AIM@Shape",

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+{
+  "name": "casting.vtu",
+  "description": "2d triangulation of the surface of a mechanical piece",
+  "acknowledgement": "",
```
"acknowledgement": "AIM@Shape",

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+  "name": "periodicPicture.cdb",
+  "description": "Cinema database containing 100 greyscale pictures with resolution 32x32, of a synthetic terrain, with 100 viewpoints arranged along a circle. Generated using ParaView",
+  "acknowledgement": "",
+  "url": ""
```
"url": "https://github.com/topology-tool-kit/ttk-data"

julien-tierny commented on README:
```diff
@@ -254,3 +254,52 @@ Data specs
+{
+  "name": "periodicPicture.cdb",
+  "description": "Cinema database containing 100 greyscale pictures with resolution 32x32, of a synthetic terrain, with 100 viewpoints arranged along a circle. Generated using ParaView",
+  "acknowledgement": "",
```
"acknowledgement": "Jules Vidal",

julien-tierny commented on docs/persistentGenerators_at.md:
```diff
@@ -0,0 +1,39 @@
+
+## C++/Python API
+
+[PersistentGenerators](https://topology-tool-kit.github.io/doc/html/classttkPersistentGenerators.html)
```
can you please make sure that **all** the TTK modules used in the corresponding python script are indeed listed here? 
(I have the impression that a few mkdocs files for this PR are missing a few TTK modules).
also, can you make sure that the entries are listed in alphabetical order?

julien-tierny commented on docs/persistentGenerators_householdAnalysis.md:
```diff
@@ -0,0 +1,53 @@
+
+## Pipeline description
+
+This example shows the detection of a circular pattern in hiigh-dimensional data.
```
hiigh --> high

julien-tierny commented on docs/persistentGenerators_periodicPicture.md:
```diff
@@ -0,0 +1,53 @@
+
+## Pipeline description
+
+This example shows the detection of a circular pattern in hiigh-dimensional data.
```
hiigh -> high

julien-tierny commented on docs/persistentGenerators_periodicPicture.md:
```diff
@@ -0,0 +1,53 @@
+
+
+## C++/Python API
+
```
back to my previous comment, the TTK cinema filters are not referenced here.
(thanks for listing all the TTK modules used in the python script).

also, thanks for putting them in alphabetical order.

julien-tierny commented on python/persistentGenerators_householdAnalysis.py:
```diff
@@ -0,0 +1,59 @@
+
+# create a new 'TTK RipsComplex'
+tTKRipsComplex1 = TTKRipsComplex(Input=tTKTableDistanceMatrix1)
+tTKRipsComplex1.SelectFieldswithaRegexp = 1
```
is the (long) specification of the input columns (below) required?
isn't the regexp sufficient?

in other words, does the script work if you remove the next line?

julien-tierny commented on python/persistentGenerators_periodicPicture.py:
```diff
@@ -0,0 +1,63 @@
+
+# create a new 'TTK TableDistanceMatrix'
+tTKTableDistanceMatrix1 = TTKTableDistanceMatrix(Input=tTKMergeBlockTables1)
+tTKTableDistanceMatrix1.InputColumns = ['0000', '0001', '0002', '0003', '0004', '0005', '0006', '0007', '0008', '0009', '0010', '0011', '0012', '0013', '0014', '0015', '0016', '0017', '0018', '0019', '0020', '0021', '0022', '0023', '0024', '0025', '0026', '0027', '0028', '0029', '0030', '0031', '0032', '0033', '0034', '0035', '0036', '0037', '0038', '0039', '0040', '0041', '0042', '0043', '0044', ...
```
cf my above remark:

"
is the (long) specification of the input columns (below) required?
isn't the regexp sufficient?

in other words, does the script work if you remove the next line?
"

julien-tierny reviewed (changes requested):
thanks a lot jules!
that's great stuff!

could you please address the few remarks I entered in my review?
thanks!

julesvidal committed:
fixed data entries in readme

julesvidal committed:
[DMS states] pass on documentation and python scripts

julesvidal commented:
Hi Julien, 
I addressed your comments. 
Just a remark : the documentation links in multiple places to https://topology-tool-kit.github.io/doc/html/classttkPersistentGenerators.html, which is not created yet. Should I remove these links ?
cheers, 
jules

julien-tierny commented:
thanks a lot jules!
(and thanks for the heads up on the dead link: I'd suggest to keep things this way for the moment, I'll update the doxygen online documentation asap)

julien-tierny commented:
awesome, let's go!

julien-tierny commented:
One more thing: 
could you please go over the source code of **each module** used in each of these examples to insert (in the paraview xml file, the vtk header, the base class header, all in alphabetical order) the link to the corresponding example pages?

here are an example for the topological simplification:
- https://github.com/topology-tool-kit/ttk/blob/dev/paraview/xmls/TopologicalSimplification.xml
- https://github.com/topology-tool-kit/ttk/blob/dev/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.h
- https://github.com/topology-tool-kit/ttk/blob/dev/core/base/topologicalSimplification/TopologicalSimplification.h

thanks!
